### PR TITLE
chore: create .gitlab-ci.yml pipeline

### PR DESCRIPTION
**Pull request by fampinheiro**

### CODE REVIEW

**Comment by jeysal** [comment]
Force pushed a config I'm happy with that does all the checks for now :+1: 
You can take over again @fampinheiro :sweat_smile:

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+image: node:12
+
+stages:
+  - setup
+  - build
+  - test
+
+setup:packages:
```
Comment (jeysal) [inline .gitlab-ci.yml:8]: this added on purpose? we already have the `check:install` step

File: package.json [hunk 2]
```diff
@@ -72,5 +73,9 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (jeysal) [inline package.json:77]: If we start distinguishing between `deps` and `devDeps` here in the workspace root, all the existing `deps` should also be `devDeps` instead (as the workspace root ofc will never have actual 'production dependencies')

Comment (fampinheiro) [reply package.json:76]: It was a consequence of using `--dev`, I can change the others.

File: packages/example-lib/src/index.ts [hunk 3]
```diff
@@ -1 +1 @@
-export const LINK_TEXT = 'React is cool.';
+export const LINK_TEXT = 'React is cool :)';
```
Comment (jeysal) [inline packages/example-lib/src/index.ts:1]: This is the key change :D

**Pull request merged**